core/src/main/java/org/mule/runtime/core/management/stats/DefaultProcessingTimeWatcher.java

```diff
@@ -58,12 +63,14 @@ public void setMuleContext(MuleContext muleContext) {
     @Override
     public void run() {
       try {
-        ProcessingTimeReference ref = (ProcessingTimeReference) queue.remove();
-        refs.remove(ref);
-
-        FlowConstructStatistics stats = ref.getStatistics();
-        if (stats.isEnabled()) {
-          stats.addCompleteFlowExecutionTime(ref.getAccumulator().longValue());
+        while (true) {
```

Comment:
If you depending on some part of your internal code to throw InterruptedException, wouldn't be better to check for interrput explicilty? What if the code you are calling at some point changes in a way that won't throw InterruptedException? For instance, it does not use any synchornization mechanism anymore